Networking/Extensions/DecodedExtension.swift

```diff
@@ -29,15 +29,21 @@ public extension Decoded {
     
 }
 
-public extension Decodable where Self: RawRepresentable, Self.RawValue == String {
+public extension Decodable where Self: RawRepresentable, Self.RawValue: Decodable {
     
     static func decode(_ json: JSON) -> Decoded<Self> {
         switch json {
-        case let .string(name): return .fromOptional(Self(rawValue: name))
-        default: return .failure(Argo.DecodeError.custom("Invalid enum value"))
+        case let .string(name) where name is Self.RawValue: return castValueToEnum(name)
+        case let .bool(value) where value is Self.RawValue: return castValueToEnum(value)
+        case let .number(value) where value is Self.RawValue: return castValueToEnum(value)
+        default: return .failure(Argo.DecodeError.custom("Invalid \(Self.self) enum value"))
         }
     }
     
+    private static func castValueToEnum(_ value: Any) -> Decoded<Self> {
```

Comment:
As it is right now no, but I could cast the values to Self.RawValue when I call the method. I left  the cast inside the method to avoid repeating it in every case. The `where value is...` clause does not actually cast the value and it doesn't seem possible to use an optional cast of the binded value as part of the check of the case, as you can do with the actual case (For example I can use `as? something` with `.string()`, but not with `name` inside the string)